src/compiler/error-detector.js

```diff
@@ -96,3 +94,27 @@ function checkExpression (exp: string, text: string, errors: Array<string>) {
     }
   }
 }
+
+function isValidExpression (exp: string) {
+  try {
+    new Function(`return ${exp}`)
+    return true
+  } catch (e) {
+    /* istanbul ignore next */
+    try {
+      const babylon = (new Function(`return require('babylon')`))()
+      /* $flow-disable-line */
+      babylon.parseExpression(exp, {
+        plugins: [
+          'objectRestSpread',
+          'functionBind',
+          'numericSeparator',
+          'optionalChaining'
+        ]
```

Comment:
@nickmessing Hmm, I don't have any ideas either for easily mapping Babel configs to babylon plugins, making the feasability of reading from `.babelrc` kind of moot anyway. 😕  Having taken a closer look at babylon now, I agree with the subset of plugins you've chosen to include and I didn't realize before that it supported ES2017 out-of-the-box. 👍